static/usage/searchbar/debounce/angular/example_component_ts.md

```diff
@@ -0,0 +1,22 @@
+```ts
+import { Component, ElementRef, ViewChildren } from '@angular/core';
+import { IonItem } from '@ionic/angular';
+
+@Component({
+  selector: 'app-example',
+  templateUrl: 'example.component.html',
+})
+export class ExampleComponent {
+  @ViewChildren(IonItem, {read: ElementRef}) items;
+
+  handleChange(event) {
+    const query = event.target.value.toLowerCase();
+
+    this.items.forEach((i) => {
+      const item = i.nativeElement;
+      const shouldShow = item.textContent.toLowerCase().indexOf(query) > -1;
+      item.style.display = shouldShow ? 'block' : 'none';
```

Comment:
Optional, but I think the example would be more impactful if this filtered out a list of data and rendered the list in a loop. Manually setting the css of the element seems a bit hacky.

So something like this:

```html
<ion-list>
  <ion-item *ngFor="result of searchResults">{{ result }}</ion-item>
</ion-list>
```

```
public dataSource = ['Amsterdam', ..., 'Panama City'];
public searchResults = [...dataSource];

handleChange(event) {
  const query = event.target.value.toLowerCase();
  this.searchResults = this.dataSource.filter(data => data.toLowerCase().indexOf(query) > -1);
}
```

Reply:
Yeah I thought about that, I was just keeping it consistent with the other examples but your way is the more Angular way so I can update it.